app/src/main/java/de/lukasneugebauer/nextcloudcookbook/recipe/util/RecipeCreateEditViewModel.kt

```
@@ -108,6 +115,51 @@ abstract class RecipeCreateEditViewModel(
+                when (val result = recipeRepository.uploadRecipeImage(image)) {
+                    is Resource.Success -> {
+                        if (result.data != null) {
+                            changeImageOrigin(result.data)
+                        }
+                    }
+                    is Resource.Error -> {
+                        _imageUploadError.value =
+                            result.message ?: UiText.StringResource(R.string.error_image_upload_failed)
+                    }
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)